### PR TITLE
fix(navbar): allows us to suppress user info in navbar with hotjar

### PR DESCRIPTION
**Pull request by MillieClare**
Currently HotJar cannot suppress user names in the navbar. This change means we can now add the suppression tag in react-libs to do so and hide all user information in screen recordings.

It's quite clunky, but `aria-label` can only take a string type, we've added a new property `aria-label` which will be set to `undefined` if it receives a label which isn't of type string.

### CODE REVIEW

File: src/components/organisms/Navbar/Navbar/Navbar.tsx [hunk 1]
```diff
@@ -22,7 +22,7 @@ import NavbarAction from '../NavbarAction/NavbarAction';
 import NavbarLinksList from '../NavbarLinksList/NavbarLinksList';
 
 export interface NavigationItem {
-  label: string;
+  label: string | React.ReactNode;
```
Comment (poteirard) [inline src/components/organisms/Navbar/Navbar/Navbar.tsx:25]: I think `React.ReactNode` has already `string` as an option so provably we can remove `string |`

Comment (poteirard) [reply src/components/organisms/Navbar/Navbar/Navbar.tsx:25]: yep
```ts
 type ReactText = string | number;
    type ReactChild = ReactElement | ReactText;

    interface ReactNodeArray extends Array<ReactNode> {}
    type ReactFragment = {} | ReactNodeArray;
    type ReactNode = ReactChild | ReactFragment | ReactPortal | boolean | null | undefined;
```

Comment (MillieClare) [reply src/components/organisms/Navbar/Navbar/Navbar.tsx:25]: Oh that's nice! I'll update now :)

File: src/components/organisms/Navbar/NavbarDropdown/NavbarDropdown.tsx [hunk 2]
```diff
@@ -78,9 +78,10 @@ export interface NavbarDropdownProps extends DefaultNavbarDropdownProps {
   /** unique id */
   id: string;
   /** dropdown label */
-  label: string;
+  label: React.ReactNode;
```
Comment (raff-r) [inline src/components/organisms/Navbar/NavbarDropdown/NavbarDropdown.tsx:81]:
```suggestion
  label: string | React.ReactNode;
```

File: src/components/organisms/Navbar/Navbar/Navbar.tsx [hunk 3]
```diff
@@ -22,7 +22,7 @@ import NavbarAction from '../NavbarAction/NavbarAction';
 import NavbarLinksList from '../NavbarLinksList/NavbarLinksList';
 
 export interface NavigationItem {
-  label: string;
+  label: React.ReactNode;
```
Comment (raff-r) [inline src/components/organisms/Navbar/Navbar/Navbar.tsx:25]:
```suggestion
  label: string | React.ReactNode;
```

File: src/components/organisms/Navbar/NavbarDropdown/NavbarDropdown.tsx [hunk 4]
```diff
@@ -148,7 +149,9 @@ export default class NavbarDropdown extends React.Component<NavbarDropdownProps,
           {label}
         </NavbarLink>
         <NavbarDropdownListContainer open={open}>
-          <NavbarDropdownList aria-label={label}>
+          {/* To avoid an undefined aria-label the fallback to label was added when aria-label property is not defined */}
+          {/* TODO remove fallback to label */}
```
Comment (admmasters) [inline src/components/organisms/Navbar/NavbarDropdown/NavbarDropdown.tsx:153]: I would make a GH issue for this so we can track the progress of fixing the issue for a future dev.

Comment (admmasters) [reply src/components/organisms/Navbar/NavbarDropdown/NavbarDropdown.tsx:153]: Other than that it looks good to me

File: src/components/organisms/Navbar/NavbarDropdown/NavbarDropdown.tsx (hunk 2 above)
Comment (poteirard) [reply src/components/organisms/Navbar/NavbarDropdown/NavbarDropdown.tsx:81]: Check the message above :P

**Review by poteirard — APPROVED** [review]

**Pull request merged**